test_framework/key.py

```diff
@@ -588,12 +611,29 @@ def tweak_add(self, tweak):
         ret.set(tweaked.to_bytes(32, 'big'), self.compressed)
         return ret
 
-def generate_schnorr_nonce():
-    """Generate a random valid bip-schnorr nonce.
+def generate_key_pair(secret=None, compressed=True):
+    """Convenience function to generate a private-public key pair."""
+    d = ECKey()
+    if secret:
+        d.set(secret, compressed)
+    else:
+        d.generate(compressed)
+
+    P = d.get_pubkey()
+    return d, P
+
+def generate_schnorr_nonce(seed=None):
+    """Generate a valid bip-schnorr nonce.
 
     See https://github.com/bitcoinops/bips/blob/v0.1/bip-schnorr.mediawiki#Signing.
     This implementation ensures the y-coordinate of the nonce point is a quadratic residue modulo the field size."""
-    kp = random.randrange(1, SECP256K1_ORDER)
+    if seed is None:
+        kp = random.randrange(1, SECP256K1_ORDER)
+    else:
+        seed = bytes(seed)
+        kp = int.from_bytes(hashlib.sha256(seed).digest(), 'big')
+        while kp > SECP256K1_ORDER:
```

Comment:
this is really unlikely, but technically bip schnorr says you should just modulo here (exactly because it's so unlikely)

Reply:
removed this functionality